oak_runtime/src/config.rs

```diff
@@ -51,11 +51,15 @@ pub fn configure_and_run(config: RuntimeConfiguration) -> Result<Arc<Runtime>, O
 }
 
 /// Load a PEM encoded TLS certificate, performing (minimal) validation.
-pub fn load_certificate(certificate: &str) -> Result<Certificate, ()> {
-    let mut cursor = std::io::Cursor::new(certificate);
+pub fn load_certificate(certificate_bytes: &[u8]) -> Result<Certificate, ()> {
+    if certificate_bytes.is_empty() {
+        error!("The certificate must be a valid non-empty byte vector");
+        return Err(());
+    }
+    let mut cursor = std::io::Cursor::new(certificate_bytes);
     // `rustls` doesn't specify certificate parsing errors:
     // https://docs.rs/rustls/0.17.0/rustls/internal/pemfile/fn.certs.html
     rustls::internal::pemfile::certs(&mut cursor)?;
 
-    Ok(Certificate::from_pem(certificate))
+    Ok(Certificate::from_pem(certificate_bytes))
```

Comment:
I haven't looked at it closely, but why is essentially the same logic present in both oak_loader and oak_runtime, with slightly different error handling?

Reply:
The checks are now moved to `oak_runtime::tls::Certificate::parse`.